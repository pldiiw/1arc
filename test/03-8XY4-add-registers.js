'use strict';

let engine = require('engine').initialize();
const addRegisters = require('instructions').addRegisters;

engine.data[0] = 20;
engine.data[1] = 240;
engine.data[2] = 30;

const engine1 = addRegisters(engine, 0, 1);
const engine2 = addRegisters(engine1, 0, 2);

console.log(engine.data[0]);
console.log(engine.data[1]);
console.log(engine.data[2]);
console.log(engine.data[0xF]);

console.log(engine1.data[0]);
console.log(engine1.data[1]);
console.log(engine1.data[2]);
console.log(engine1data[0xF]);

console.log(engine2.data[0]);
console.log(engine2.data[1]);
console.log(engine2.data[2]);
console.log(engine2.data[0xF]);
