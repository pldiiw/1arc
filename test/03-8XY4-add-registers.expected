20
240
30
0
4
240
30
1
34
240
30
0
